saleor/graphql/discount/mutations/sale/sale_delete.py

```diff
@@ -43,22 +48,27 @@ class Meta:
     def perform_mutation(  # type: ignore[override]
         cls, root, info: ResolveInfo, /, *, id: str
     ):
-        instance = cls.get_node_or_error(info, id, only_type=Sale)
-        previous_catalogue = fetch_catalogue_info(instance)
-        manager = get_plugin_manager_promise(info.context).get()
+        object_id = cls.get_global_id_or_error(id, "Sale")
+        promotion = models.Promotion.objects.get(old_sale_id=object_id)
+        promotion_id = promotion.id
+        rules = promotion.rules.all()
```

Comment:
isn't better to fetch only a `rules[0]` instead of all of them? 
In all places the `rule[0]` is used. Is there any reason why it is done this way?

Reply:
good point, thanks!